src/components/Messaging.tsx

```diff
@@ -1,172 +1,111 @@
-
-        this.localConnection
-            .createOffer()
-            .then((offer: RTCSessionDescriptionInit) => this.localConnection.setLocalDescription(offer))
-            // TODO: send signal over server
-            .then(() =>
-                this.remoteConnection.setRemoteDescription(this.localConnection
-                    .localDescription as RTCSessionDescription),
-            )
-            .then(() => this.remoteConnection.createAnswer())
-            .then((answer: RTCSessionDescriptionInit) => this.remoteConnection.setLocalDescription(answer))
-            .then(() =>
-                this.localConnection.setRemoteDescription(this.remoteConnection
-                    .localDescription as RTCSessionDescription),
-            )
-            .catch(this.handleCreateDescriptionError);
-    };
-
-    disconnectPeers = () => {
-        (this.sendChannel as RTCDataChannel).close();
-        (this.receiveChannel as RTCDataChannel).close();
-
-        this.localConnection.close();
-        this.remoteConnection.close();
-
-        this.sendChannel = null;
-        this.receiveChannel = null;
-        this.localConnection = new RTCPeerConnection();
-        this.remoteConnection = new RTCPeerConnection();
+    /**
+     * Disconnect from RTC channels. The cool thing here is that if ANY user in the room leaves,
+     * all other users will re-render their components and since room is not full, they will also be disconnected.
+     */
+    componentWillUnmount(): void {
+        RTC.disconnect();
+    }
 
-        this.setState({
-            connectButtonDisabled: false,
-            disconnectButtonDisabled: true,
-            messageInputDisabled: true,
-            messageInputText: '',
-        });
-    };
+    /**
+     * Custom handler for status change on send channel. Needed to re-render component.
+     */
+    handleSendChannelStatusChange(open: boolean): void {
+        this.setState({ sendChannelOpen: open })
+    }
 
-    handleSendChannelStatusChange = (event: Event) => {
-        if (this.sendChannel) {
-            let state = this.sendChannel.readyState;
+    /**
+     * Custom handler for receive change on send channel. Needed to re-render component.
+     */
+    handleReceiveChannelStatusChange(open: boolean): void {
+        this.setState({ receiveChannelOpen: open });
+    }
 
-            if (state === 'open') {
-                this.setState({
-                    connectButtonDisabled: true,
-                    disconnectButtonDisabled: false,
-                    messageInputDisabled: false,
-                });
-            } else {
-                this.setState({
-                    connectButtonDisabled: false,
-                    disconnectButtonDisabled: true,
-                    messageInputDisabled: true,
-                });
-            }
+    /**
+     * Handles messages sent over RTC send channel.
+     */
+    handleSendMessage(msg: string): void {
+        if (RTC.sendChannel) {
```

Comment:
Do you think it'd be better to move this if statement into the RTC module? Something like
```
handleSendMessage(msg: string): void {
  if (RTC.sendMessage(msg)) {
    this.addMessage(msg);
  }
}
```
And inside `lib/RTC.ts`
```
class RTC {
  // constructor and methods ...
  sendMessage(msg: string): boolean {
    if (this.sendChannel) {
      this.sendChannel.send(msg);
      return true;
    }
    return false;
  }
}
```

Reply:
Yea, good call.